filament/backend/src/metal/MetalHandles.mm

```diff
@@ -111,7 +111,12 @@ static inline MTLStorageMode getMetalStorageMode(TextureUsage usage) {
     if (size <= 4 * 1024) {   // 4K
         bufferPoolEntry = nullptr;
         cpuBuffer = malloc(size);
+        return;
     }
+    // To satisfy Metal, We need to preallocate a buffer in case Filament decides to draw before
```

Comment:
Could this allocation be deferred till later, like during the bind time or the draw?  Would be cool if we could somehow remove the new "preallocated" bool and simply check for the nullness of `bufferPoolEntry` at a later point.

Reply:
Good suggestion